### PR TITLE
Contest mode now works very differently, and waits for the contest to start

### LINKED ISSUE #69: If contest get is used before contest opens, it should wait until the contest opens and then get the problems

### PR DESCRIPTION
**Pull request by Jakobis**
Fixes #69 Will need some additional testing at next lillekat

### CODE REVIEW

File: commands/archive.py [hunk 1]
```diff
@@ -1,5 +1,7 @@
 from argparse import ArgumentParser
 import os, shutil
+
+from helpers.types import problem, problemList
 from helpers.webutils import promptToFetch
```
Comment (Hjaltesorgenfrei) [inline commands/archive.py:5]: Sort imports

File: commands/archive.py [hunk 2]
```diff
@@ -11,6 +13,7 @@ def archiveCommand(data):
             archive(problem)
 
 def archive(problemName, folder=".archive/"):
+    print(problemName)
```
Comment (Hjaltesorgenfrei) [inline commands/archive.py:16]: Is this a debug print?

File: commands/contest.py [hunk 3]
```diff
@@ -1,52 +1,79 @@
...
 from argparse import ArgumentParser
-from sys import stderr
 from helpers.timeutils import toDatetime, toTimeDelta
 from commands.submit import submitCommand
 from helpers.cli import yes
 
 from bs4 import BeautifulSoup
-from helpers.config import getConfigUrl
-import re
+from helpers.config import getConfig, saveConfig
```
Comment (Hjaltesorgenfrei) [inline commands/contest.py:8]: Sort imports

File: commands/contest.py [hunk 4]
```diff
@@ -71,34 +106,21 @@ def readContest(contest, session):
 
     return {
         'problems': problems,
-        'inProgress': ((not timeTo) or timeTo <= toTimeDelta("0:00:00")) and (remaining != toTimeDelta("0:00:00")),
+        'problemMap': {string.ascii_lowercase[i]: x for i, x in enumerate(problems)},
+        'timeState': timeState,
+
```
Comment (Hjaltesorgenfrei) [inline commands/contest.py:111]: Empty line

File: commands/get.py [hunk 5]
```diff
@@ -1,5 +1,7 @@
 from argparse import ArgumentParser
 import subprocess
+
+from helpers.types import problem, problemList
 from helpers.programSelector import formatCommand, selectProgramFile
 from helpers.config import getConfig
 from commands.web import webCommand
```
Comment (Hjaltesorgenfrei) [inline commands/get.py:7]: Sort imports

**Review by Hjaltesorgenfrei — CHANGES_REQUESTED** [review]

File: commands/contest.py [hunk 6]
```diff
@@ -1,52 +1,79 @@
...
+                return
+    solved = getCommand({
+        **data,
+        'command': 'get',
+        'problem': contestData.get('problems'),
+    })
+    #update config with new problems
+    config = getConfig()
+    config['contest'] = contestData['problemMap']
+    saveConfig()
```
[the remark this answers is not in the document]
Comment (Duckapple) [reply commands/contest.py:54]: oh well, I guess it works lol.

**Review by Duckapple — APPROVED** [review]

**Pull request merged**